https://github.com/libffi/libffi/commit/9c9e8368e49804c4f7c35ac9f0d7c1d0d533308b

From 9c9e8368e49804c4f7c35ac9f0d7c1d0d533308b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 4 Jun 2024 14:13:08 +0300
Subject: [PATCH] aarch64: Add a missing no-op define of SIGN_LR_LINUX_ONLY
 (#838)

This is needed at least if building for Linux, with a toolchain
that doesn't default to having PAC enabled, fixing build errors
since 45d284f2d066cc3a080c5be88e51b4d934349797.
---
 src/aarch64/internal.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/aarch64/internal.h b/src/aarch64/internal.h
index c39f9cb2..50fa5c13 100644
--- a/src/aarch64/internal.h
+++ b/src/aarch64/internal.h
@@ -88,6 +88,7 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
   #define AUTH_LR_AND_RET retab
   #define AUTH_LR_WITH_REG(x) autib lr, x
   #define BRANCH_AND_LINK_TO_REG blraaz
+  #define SIGN_LR_LINUX_ONLY
   #define BRANCH_TO_REG braaz
   #define PAC_CFI_WINDOW_SAVE
   /* Linux PAC Support */
@@ -136,6 +137,7 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
     #define AUTH_LR_AND_RET ret
     #define AUTH_LR_WITH_REG(x)
     #define BRANCH_AND_LINK_TO_REG blr
+    #define SIGN_LR_LINUX_ONLY
     #define BRANCH_TO_REG br
     #define PAC_CFI_WINDOW_SAVE
   #endif /* HAVE_ARM64E_PTRAUTH */

